modules/foreman_provisioning_template.py

```diff
@@ -0,0 +1,404 @@
+      name: timeywimey_template
+      template: |
+        <%#
+            dummy:
+        %>
+      state: absent
+
+- name: "Create a Provisioning Template from a file and modify with parameter"
+  local_action:
+      module: foreman_provisioning_template
+      username: "admin"
+      password: "admin"
+      server_url: "https://foreman.example.com"
+      file_name: timeywimey_template.erb
+      name: Wibbly Wobbly Template
+      state: present
+      locations:
+      - Gallifrey
+      organizations:
+      - TARDIS INC
+
+# Providing a name in this case wouldn't be very sensible.
+# Alternatively make use of with_filetree to parse recursively with filter.
+- name: "Parsing a directory of provisioning templates"
+  local_action:
+      module: foreman_provisioning_template
+      username: "admin"
+      password: "admin"
+      server_url: "https://foreman.example.com"
+      file_name: "{{ item }}"
+      state: present
+      locations:
+      - SKARO
+      organizations:
+      - DALEK INC
+      with_fileglob:
+       - "./arsenal_templates/*.erb"
+
+'''
+
+RETURN = ''' # '''
+
+
+try:
+    from nailgun.entities import (
+        ProvisioningTemplate,
+        OperatingSystem,
+        _OPERATING_SYSTEMS,
+    )
+
+    from ansible.module_utils.ansible_nailgun_cement import (
+        TemplateKind,
+        Organization,
+        Location,
+        create_server,
+        find_entity,
+        find_entities,
+        create_entity,
+        update_entity,
+        parse_template,
+        parse_template_from_file,
+    )
+
+    HAS_NAILGUN_PACKAGE = True
+except ImportError:
+    HAS_NAILGUN_PACKAGE = False
+
+import os
+from ansible.module_utils.basic import AnsibleModule, get_module_path
+from ansible.module_utils._text import to_bytes, to_native
+
+
+def find_template_kind(template_dict, module):
+    if 'kind' not in template_dict:
+        module.fail_json(msg='Could not infer template kind!')
+
+    template_dict['snippet'] = (template_dict['kind'] == 'snippet')
+    if template_dict['snippet']:
+        template_dict.pop('kind')
+    else:
+        try:
+            template_dict['kind'] = find_entity(
+                TemplateKind, name=template_dict['kind'])[0]
+        except Exception as e:
+            module.fail_json(msg='Template kind not found!')
+    return template_dict
+
+
+def sanitize_template_dict(template_dict):
+    # This is the only true source for names (and conversions thereof)
+    name_map = {
+        'audit_comment': 'audit_comment',
+        'kind': 'template_kind',
+        'locations': 'location',
+        'locked': 'locked',
+        'name': 'name',
+        'organizations': 'organization',
+        'snippet': 'snippet',
+        'template': 'template',
+    }
+    result = {}
+    for key, value in name_map.iteritems():
+        if key in template_dict:
+            result[value] = template_dict[key]
+    return result
+    # Missing parameters:
```

Comment:
This is just a reminder, that these parameters are not yet implemented in nailgun.
We would like to keep the comments, until this is resolved.